xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/internal/macro/AbstractShowHideIfMacro.java

```diff
@@ -59,6 +71,9 @@ public abstract class AbstractShowHideIfMacro extends AbstractProMacro<ShowHideI
     @Inject
     private GroupManager groupManager;
 
+    @Inject
+    private RenderingScriptService renderingScriptService;
```

Comment:
@Josue-T 

Should have been
```
    @Inject
    @Named("rendering")
    private ScriptService renderingScriptService;
```
And then cast it to RenderingScriptService when using it.

It is hackish, but I'm not sure we have other solutions right now.

You should also add a TODO comment to make use of `XWikiSyntaxEscaper` after upgrading the parent of the app to something greater than 14.10.6

Reply:
Thank you a lot @trrenty, if work this way.